contracts/RepaymentController.sol

```diff
@@ -63,33 +66,75 @@ contract RepaymentController is IRepaymentController, InterestCalculator, FeeLoo
     // ==================================== LIFECYCLE OPERATIONS ========================================
 
     /**
-     * @notice Repay an active loan, referenced by borrower note ID (equivalent to loan ID). The interest for a loan
-     *         is calculated, and the principal plus interest is withdrawn from the caller.
-     *         Anyone can repay a loan. Control is passed to LoanCore to complete repayment.
+     * @notice Send a repayment for an active loan, referenced by BorrowerNote ID (equivalent to loan ID).
+     *         The interest for the loan is calculated, and the repayment amount plus interest is withdrawn
+     *         from the caller. The repayment amount at a minimum must cover any interest accrued. There are
+     *         no partial repayments to interest, only principal. Anyone can repay a loan. After the
+     *         repayment amounts are calculated, control is passed to LoanCore to complete repayment and
+     *         update LoanData accounting.
      *
      * @param  loanId               The ID of the loan.
+     * @param  amount               The amount to repay.
      */
-    function repay(uint256 loanId) external override {
-        (uint256 amountFromBorrower, uint256 amountToLender) = _prepareRepay(loanId);
-
-        // call repay function in loan core -  msg.sender will pay the amountFromBorrower
-        loanCore.repay(loanId, msg.sender, amountFromBorrower, amountToLender);
+    function repay(uint256 loanId, uint256 amount) external override {
+        // no zero amount check, minimum principal could be 0 for this payable currency
```

Comment:
Hey, good q. There is a scenario in the tests where a loan could be started with a zero amount principal. This can potentially occur because in the `setAllowedPayableCurrencies` fn there is no ZeroAmount check on the `currencyData[i].minPrincipal` input. So it is possible to have 0 principal loans, even none of the currently whitelisted currencies in V3 support this because there minimums are higher than zero, it is a possibility. The test for this can be found in the RepaymentController.ts file titled "repays a loan with 0 amount due"